query/executor.go

```diff
@@ -289,6 +306,94 @@ func (e *Executor) WithLogger(log *zap.Logger) {
 	e.TaskManager.Logger = e.Logger
 }
 
+func startFileLogWatcher(w WatcherInterface, e *Executor, ctx context.Context) error {
```

Comment:
`startFileLogWatcher` made me think this was going to spawn a goroutine, but it doesn't. It's the body of a goroutine. Maybe either `doFileLogWatch` or simply `fileLogWatch`?